### PR TITLE
[BE] feat: Implement checking profile follows another profile

### PR DESCRIPTION
**Pull request by francosilva25**
## ✏️ Follow Profile - Check Following Status

The Follow Profile - Check Following Status API endpoint allows the client to determine whether one profile is following another on the social media platform. This information is valuable for user interactions and relationship tracking.

### **API Overview**

The API utilizes the HTTP GET method to retrieve the following status of a specific profile.

### **Endpoint**

Access the Check Following Status service through the following endpoint:

`http://localhost:3333/follow_profile/is_following`

### **Request**

The client needs to provide the following parameters in the request:

- `follower_profile_id` (integer): The unique identifier of the profile that is potentially following.
- `following_profile_id` (integer): The unique identifier of the profile that is potentially being followed.

**Example Request:**

```http
GET http://localhost:3333/follow_profile/is_following?follower_profile_id=1&following_profile_id=2
```

### **Response**
The server responds with a JSON object containing the following information:

- follower_profile_id (integer): The unique identifier of the follower profile.
- following_profile_id (integer): The unique identifier of the profile being followed.
- is_following (boolean): Indicates whether the follower is following the specified profile.

**Example Response:**
```json
{
  "follower_profile_id": 1,
  "following_profile_id": 2,
  "is_following": true
}
```

### CODE REVIEW

File: backend/internal/follow/application/followUseCase.go [hunk 1]
```diff
@@ -38,3 +39,20 @@ func (r *FollowerUseCase) FollowProfile(p_follower_profile_id, p_following_profi
 	}
 	return newFollower, nil
 }
+
+func (r *FollowerUseCase) IsFollowing(p_follower_profile_id, p_following_profile_id int) (*bool, error) {
```
Comment (ekimike) [inline backend/internal/follow/application/followUseCase.go:43]: In which scenario this test fails?

File: backend/internal/follow/infrastructure/followRepository.go [hunk 2]
```diff
@@ -33,3 +33,42 @@ func (u *FollowerRepository) InsertNewFollower(newFollower *domain.Follower) (*d
...
+
+	rows, err := tx.Query(query, newFollower.Follower_profile_id, newFollower.Following_profile_id)
+	if err != nil {
+		log.Println("Error while executing the query")
+		return nil, err
+	}
+	defer rows.Close()
+
+	var isFollowing bool
+	for rows.Next() {
```
Comment (ekimike) [inline backend/internal/follow/infrastructure/followRepository.go:57]: Does this query return more than one row?

File: backend/internal/follow/application/followUseCase.go (hunk 1 above)
Comment (francosilva25) [reply backend/internal/follow/application/followUseCase.go:43]: Good evening, Professor. According to what we discussed, more cases were added to the unit tests.

File: backend/internal/follow/infrastructure/followRepository.go (hunk 2 above)
Comment (francosilva25) [reply backend/internal/follow/infrastructure/followRepository.go]: Good evening professor.

The query only returns one row, the row traversal has been solved.

**Review by miguel1808 — APPROVED** [review]
I see that you have solved the teacher's queries by correcting the code and making it more optimal.

**Pull request merged**